google/cloud/pubsub/quickstart/subscriber_quickstart.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! [START pubsub_quickstart_subscriber] [all]
+#include "google/cloud/pubsub/message.h"
+#include "google/cloud/pubsub/subscriber.h"
+#include <iostream>
+
+int main(int argc, char* argv[]) try {
+  if (argc != 3) {
+    std::cerr << "Usage: " << argv[0] << " <project-id> <subscription-id>\n";
+    return 1;
+  }
+
+  std::string const project_id = argv[1];
+  std::string const subscription_id = argv[2];
+
+  // Create a namespace alias to make the code easier to read.
+  namespace pubsub = ::google::cloud::pubsub;
+
+  auto subscriber = pubsub::Subscriber(pubsub::MakeSubscriberConnection(
+      pubsub::Subscription(project_id, subscription_id)));
+
+  auto session =
+      subscriber
+          .Subscribe([&](pubsub::Message const& m, pubsub::AckHandler h) {
+            std::cout << "Received message " << m << std::endl;
+            std::move(h).ack();
+          })
+          .then([](auto f) { throw f.get(); });
```

Comment:
Are you certain this is what you want to do?

Reply:
No...

I want it to throw an exception if the Pub/Sub service returns one.

Added a check to see when the session completes if the status is ok or not.